src/resmom/linux/alps.c

```diff
@@ -799,12 +799,10 @@ stack_busted(ud_t *d)
 			}
 		} else if (strcmp(BASIL_ELM_RSVNARRAY, top) == 0) {
 			if (strcmp(BASIL_ELM_INVENTORY, prev) != 0) {
-				parse_err_illegal_start(d);
-				return (1);
-			}
-			if (brp->data.query.type != basil_query_inventory) {
-				parse_err_illegal_start(d);
-				return (1);
+				if (strcmp(BASIL_ELM_RESPONSEDATA, prev) != 0) {
+					parse_err_illegal_start(d);
+					return (1);
+				}
```

Comment:
This is wrong, we cannot check for the query type in stack_busted(), because brp->data.query.type doesn't get set until we start parsing the values within the XML.

Reply:
That's correct Lisa, I don't know what I was thinking. You are right this will fail, I don't see any way to work around it because in stack busted we don't know which query type we are handling. I shall take this off.

Reply:
@lisa-altair are you certain that is the case? When I look at the code prior to these changes, I can clearly see a test being performed on data.query.type. Are you saying the existing code was incorrect?

Reply:
@mike0042 you're right, it could be checked.  But wouldn't that be verifying the order in which the PBS code ran (because we expected basil_query_status to be set), as opposed to verifying the structure of the XML elements?

Reply:
I see that a check for "data.query.type" is there at line 694 but I am confused as how it is useful.
we check for node array element to come after Inventory and if it is not inventory the we throw an error. But, if the check passes then we check for data.query.type to be anything other than inventory, which will be false no matter what, because "data.query.type" is set in inventory_start.

Reply:
That check was there for a reason. A reason that may no longer apply. I think my concern is based on some fuzzy memories of how the code used to work long ago. I don't see a reason for holding this back any longer, and I can't find a good reason to change what you currently have in the PR.